main.tf

```
@@ -111,6 +111,15 @@ resource "azurerm_storage_account" "this" {
     bypass                     = !var.network_rules_bypass_azure_services ? [] : ["AzureServices"]
     ip_rules                   = var.network_rules_ip_rules
     virtual_network_subnet_ids = var.network_rules_virtual_network_subnet_ids
+
+    dynamic "private_link_access" {
+      for_each = var.endpoint_resource_id != null ? [1] : []
+
+      content {
+        endpoint_resource_id = var.endpoint_resource_id
+        endpoint_tenant_id   = var.endpoint_tenant_id
+      }
+    }
```

You can't have a conditional based on a resource ID, since a resource ID is not known until the end of the apply phase, and conditionals must be based on values known during the planning phase.

Also, according to the wording in the documentation for the `azurerm_storage_account` resource, you can configure one or more blocks, so we need to open up the possibility of configuring multiple blocks of this type.

```suggestion
    dynamic "private_link_access" {
      for_each = var.private_link_accesses

      content {
        endpoint_resource_id = private_link_access.value.endpoint_resource_id
        endpoint_tenant_id   = private_link_access.value.endpoint_tenant_id
      }
    }
```